src/bandersnatch/package.py

```diff
@@ -40,50 +28,30 @@ def __str__(self) -> str:
 
 
 class Package:
-    def __init__(self, name: str, serial: Union[int, str], mirror: "Mirror",) -> None:
+    def __init__(self, name: str, master: Master, serial: int = 0) -> None:
         self.name = canonicalize_name(name)
         self.raw_name = name
         self.serial = serial
-        self.mirror = mirror
+        self.master = master
 
-    @property
-    def json_file(self) -> Path:
-        return Path(self.mirror.webdir / "json" / self.name)
+        self._metadata = None
 
     @property
-    def json_pypi_symlink(self) -> Path:
-        return Path(self.mirror.webdir / "pypi" / self.name / "json")
+    def metadata(self) -> dict:
+        assert self._metadata is not None, "Must fetch metadata before accessing it"
```

Comment:
asserts in optimized python won't be evaluated at runtime ...

Reply:
What do you mean? I don't think Python ever skips asserts (unlike languages like C, Rust)

Reply:
So I've just learned about python's ```-O``` option, but I don't think it's commonly used because it does pretty much nothing.

Reply:
Not a fan, but can live with it. I feel we should raise a `NeedMetadataError` from your new errors.py eventually.